legal-api/src/legal_api/reports/report.py

```diff
@@ -216,6 +223,24 @@ def _format_directors(self, directors):
                 pass
         return directors
 
+    def _set_translations(self, filing):
+        # initial (existing) translations, if any
+        name_translations = filing['business'].get('nameTranslations', [])
```

Comment:
@lekshmimallika-aot I'm assuming there will be a list of existing name translations in the business object. Is this actually the plan? If not, where do I get the initial list of name translations for an alteration?

Reply:
For which report is this? NOA ?

Reply:
Technically, this is for "not IA" filings -- so yes, for Notice of Alteration.

Reply:
NOA is available only after filing is complete. So you can directly get it from the business object and set it to the json. Its not just for name translations. director addresses etc wont be captured in the schema for alteration yet they have to be present in the output. The same logic applies here

Reply:
That makes sense, thanks.

Do you know if the business object is complete and up to date, or should it be queried from the db? For example, if this is a COA/COD filing, will its business object contain the current name translations?

I figure this code will be used for NOA report for an alteration filing. Does this make sense?

Reply:
> Do you know if the business object is complete and up to date, or should it be queried from the db? For example, if this is a COA/COD filing, will its business object contain the current name translations?

This part will need to be updated - see /bcgov/entity#4299.

Reply:
> Technically, this is for "not IA" filings -- so yes, for Notice of Alteration report.

seems like you guys are talking about 2 different things here. A NOA (Notice of Articles) is available after the filing is complete. A Notice of Alteration is a Filing and is available upon submission.

Reply:
Oops, typo. I meant Notice of Articles... fixed in comment above. Thanks for catching that.

I think we are trying to call the Alteration filing just that -- not "Notice of..." -- to reduce confusion :)